lib/MusicBrainz/Server/Data/ReleaseGroup.pm

```diff
@@ -157,6 +158,32 @@ sub has_materialized_artist_release_group_data {
     return $has_data;
 }
 
+sub load_has_cover_art {
+    my ($self, @release_groups) = @_;
+    my %id_to_rg = object_to_ids(@release_groups);
+    my @ids = keys %id_to_rg;
+
+    return unless @ids; # nothing to do
+
+    my $query = <<~'SQL';
+        SELECT release.release_group
+          FROM release
+          JOIN cover_art_archive.cover_art ca ON ca.release = release.id
+          JOIN cover_art_archive.cover_art_type cat ON cat.id = ca.id
+          JOIN release_meta ON release_meta.id = release.id
+         WHERE release.release_group = any(?)
+           AND cat.type_id = 1
+           AND ca.mime_type != 'application/pdf'
+           AND release_meta.cover_art_presence != 'darkened'
+        SQL
+
+    my $ids_with_art = $self->sql->select_single_column_array($query, \@ids);
+
+    for my $id (@{ $ids_with_art }) {
+        $id_to_rg{$id}->[0]->has_cover_art(1);
```

Comment:
`object_to_ids` groups them in arrays in case you have multiple instances of the same entity for some reason -- looking at the current places we use `load_has_cover_art`, this might happen on series pages?
```suggestion
        for my $rg in (@{ $id_to_rg{$id} }) {
            $rg->has_cover_art(1);
        }
```